### PR TITLE
Add Preset Dropdown to Blender exporter 

### PR DESCRIPTION
**Pull request by rkusa**
Suggested by @satori99 in #6177. Adds the following dropdown to the exporter:

screen shot 2015-03-05 at 07 23 17

### CODE REVIEW

**Comment by rkusa** [comment]
@repsac If this gets merged, we could probably remove `save_settings_export` and `restore_settings_export` from the exporter, couldn't we?

**Comment by repsac** [comment]
I don't see why not.

**Comment by repsac** [comment]
Do you want to move forward with this PR or remove the old settings logic all together?

File: utils/exporters/blender/addons/io_three/exporter/api/mesh.py [hunk 1]
```diff
@@ -46,7 +46,7 @@ def skeletal_animation(mesh, options):
         logger.warning("No armature found (%s)", mesh)
         return []
 
-    anim_type = options.get(constants.ANIMATION)
+    anim_type = options.option_animation_skeletal
```
Comment (rkusa) [inline utils/exporters/blender/addons/io_three/exporter/api/mesh.py]: Is `option_animation_skeletal` correct here? There was no `option_animation`.

**Comment by rkusa** [comment]
I've added a commit that removes the obsolete setting handling.  But I am not sure about this commit, because I had to replace, e.g.:

```
maps = options.get(constants.MAPS)
```

with

```
maps = options.option_maps
```

@repsac What do you think about these change?

**Comment by repsac** [comment]
Uh, I don't remember getting seeing an email with your reply. Sorry for the late reply. Looking at this now.

**Comment by repsac** [comment]
OK so the first thing I am concerned about is `options` becoming an object with properties. That breaks a lot of things, including all of my [test scripts] (https://github.com/mrdoob/three.js/tree/master/utils/exporters/blender/tests/scripts) as well as some other batch scripts I use.

`options` being a dictionary allows a user to easily utilize `io_three.exporter`, without depending on the actual addon itself being used, from any type of custom script. Refer to [exporter.py](https://github.com/mrdoob/three.js/blob/master/utils/exporters/blender/tests/scripts/exporter.py) as an example.
I think using `vars()` and converting `options` to a dictionary would be a possibility.
```
>>> foo = type('Foo', (), {'x': 1, 'y': 2})
>>> foo.x
1
>>> bar = vars(foo)
>>> bar['x']
1
```

however that means addressing the naming of the various controls so they match keys in the constants (removing the "option_" prefix"). Let me think a lot of this over on how else to approach this.

File: utils/exporters/blender/addons/io_three/__init__.py [hunk 2]
```diff
@@ -257,216 +257,6 @@ def draw(self, context):
...
-        constants.EMBED_ANIMATION: properties.option_embed_animation,
-        constants.LIGHTS: properties.option_lights,
-        constants.CAMERAS: properties.option_cameras,
-
-        constants.MORPH_TARGETS: properties.option_animation_morph,
-        constants.ANIMATION: properties.option_animation_skeletal,
-        constants.FRAME_STEP: properties.option_frame_step,
-        constants.FRAME_INDEX_AS_TIME: properties.option_frame_index_as_time,
-        constants.INFLUENCES_PER_VERTEX: properties.option_influences
-    }
```
Comment (rkusa) [inline utils/exporters/blender/addons/io_three/__init__.py]: @repsac We could manually populate a dictionary. Should be just copy and paste of this snippet above, shouldn't it?

Comment (repsac) [inline utils/exporters/blender/addons/io_three/__init__.py]: Yeah, check this branch out and let me know if this works. It only edits the __init__.py and doesn't touch rest of the modules and doesn't break the test cases.
https://github.com/repsac/three.js/tree/bl_options

Apologize in advance. You had to edit a lot of files for this PR.

**Comment by rkusa** [comment]
@repsac your `bl_options` branch looks good! Thanks for having a look and adapting the changes. Don't worry about the files that had to be changed for the PR - regex find and replace did most of the job :smiley:

**Comment by repsac** [comment]
OK cool. Didn't want to make you feel your effort wasn't respected. I noticed you uncommented the option for embedded geometry, is that supported in ObjectLoader now?

**Comment by rkusa** [comment]
I had to uncomment it, because I needed the `option_embed_geometry` property to replace the dictionary get call. So, no ObjectLoader support now

**Comment by repsac** [comment]
Thanks for the clarification. Will setup a new PR.

**Pull request closed without merging**